contracts/CryptoMafiaCoin.sol

```diff
@@ -4,12 +4,45 @@ pragma solidity ^0.8.0;
 import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
 
+interface RandomInterface {
+  function getRandomNumber(uint256 userProvidedSeed) external returns (bytes32);
+  function randomNumber() external view returns(uint256);
+}
+
+interface AccountInterface{
+    function ownerOf(uint256 tokenId) external view returns (address);
+}
+
 contract CryptoMafiaCoin is ERC20, Ownable {
     address private _accountsHolder;
-
-    constructor(address accountsHolder, uint256 initialSupply) ERC20("Crypto Mafia Coin", "CMC") {
+    
+    address addressRandomGenerator;
+    
+    address contractAddress;
+    
+    RandomInterface RNG;
+    
+    AccountInterface AI;
+    
+    mapping(uint256=>uint256) roomStake;
+    mapping(uint256=>mapping(uint256=>address)) playerID;
+    mapping(uint256=>uint256) roomRandomNumber;
+    mapping(uint256=>bool) randomNumIsTaken;
```

Comment:
No need. if first mapping is returning 0 then random number haven't fullfilled

Reply:
If we check key value mapping for just if a value exists for the key that doesn't guarantee that the value would be unique, basically I am check if a random number is already in use or not

Reply:
We don't need unique random no. we just need a random no. for roomId. It's okay if sometime 2 random no. are equal